main.go

```diff
@@ -510,21 +514,25 @@ is available in the $BITRISE_IDEDISTRIBUTION_LOGS_PATH environment variable`)
 		}
 	}
 
-	if err := utils.ExportOutputFile(exportedIPAPath, exportedIPAPath, bitriseIPAPthEnvKey); err != nil {
+	if err := utils.ExportOutputFile(commandFactory, exportedIPAPath, exportedIPAPath, bitriseIPAPthEnvKey); err != nil {
 		fail("Failed to export %s, error: %s", bitriseIPAPthEnvKey, err)
 	}
 
 	log.Donef("The ipa path is now available in the Environment Variable: %s (value: %s)", bitriseIPAPthEnvKey, exportedIPAPath)
 
-	appDSYM, _, err := archive.FindDSYMs()
+	appDSYMs, _, err := archive.FindDSYMs()
 	if err != nil {
 		fail("Failed to export dsym, error: %s", err)
 	}
 
-	if err := utils.ExportOutputDirAsZip(appDSYM, dsymZipPath, bitriseDSYMPthEnvKey); err != nil {
+	if len(appDSYMs) == 0 {
+		log.Warnf("No dSYM was found in the archive")
+		return
+	}
+
+	if err := utils.ExportOutputDirAsZip(commandFactory, appDSYMs[0], dsymZipPath, bitriseDSYMPthEnvKey); err != nil {
```

Comment:
Previously the `archive.FindDSYMs()` call was only returning a single app dsym path string and after the upgrade this is an array with potentially multiple dsyms.

The `utils.ExportOutputDirAsZip()` function has a hidden side effect which is not visible from here. It will take the supplied filepath and only use the parent directory from it. All the dSYMs are in the `dSYMs` folder inside the `xcarchive`. So what this line actually means is: `zip the dSYMs folder from the xcarchive and export it.`

That is why using the first item is fine here.

Reply:
`findDSYMs` searches for DSYM files in `<archivePath>/dSYMs` dir (this is the parent dir of each DSYM file).
This dir contains both app and framework DSYMs.
We intentionally distinguish app and framework DSYMs.
With this change, we will alter the current behavior and export both app and framework DSYMs too.

Reply:
We already export everything from the `<archivePath>/dSYMs` folder. 

In the previous behaviour `archive.FindDSYMs()` returned a single app dsym path string and an array of framework dsym path string. That single app dsym path was fed to the `utils.ExportOutputDirAsZip()` call. 

After the util update the `archive.FindDSYMs()` returns also an array of app dsyms paths instead of a single one. I just take the first item and feed it to the `utils.ExportOutputDirAsZip()` call. 

The only behaviour change I introduced is that if there are no app dsym paths in the returned array then it prints a warning. This should never really happen as there always should be at least one dysm for the app.